src/main/java/pl/umk/mat/git2befit/service/workout/factory/implementation/SplitTrainingPlan.java

```diff
@@ -17,68 +19,71 @@
     private final List<String> smallBodyParts = List.of("SIXPACK", "CALVES", "BICEPS", "TRICEPS", "SHOULDERS");
     private final List<String> bigBodyParts = List.of("CHEST", "BACK", "THIGHS");
 
-    private final ExerciseRepository exerciseRepository;
-
+    private List<Exercise> exercisesWithoutEquipment = new ArrayList<>();
+    private List<Exercise> exercisesWithEquipment = new ArrayList<>();
     private TrainingForm trainingForm;
+    private TrainingForm localTrainingForm;
 
+    private final ExerciseRepository exerciseRepository;
 
     public SplitTrainingPlan(ExerciseRepository exerciseRepository) {
         this.exerciseRepository = exerciseRepository;
     }
 
-
-    @Override
-    public List<Training> create(TrainingForm trainingForm) {
+    private void initialize(TrainingForm trainingForm) {
         this.trainingForm = trainingForm;
-        List<Exercise> exerciseListFilteredByTrainingType = getExerciseListFilteredByTrainingType();
-        List<Exercise> exercisesWithoutEquipment = exerciseRepository.getAllWithNoEquipmentForTrainingTypeName(TRAINING_TYPE);
-        List<Exercise> exercisesWithEquipment = filterAllByAvailableEquipment(exerciseListFilteredByTrainingType, trainingForm.getEquipmentIDs());
+        this.localTrainingForm = trainingForm;
 
-        Map<String, List<ExerciseExecution>> trainingForBodyPart = assignExercisesToBodyPart(exercisesWithEquipment, exercisesWithoutEquipment);
+        List<Exercise> exerciseListFilteredByTrainingType = exerciseRepository.getAllByTrainingTypes_Name(TRAINING_TYPE);
+        BooleanSupplier isContainsEquipmentWithout = () -> trainingForm.getEquipmentIDs().contains(20L);
```

Comment:
Chcąc stosować określenia w stronie biernej czasownik musi być w czasie przeszłym, tzn ```isContainsEquipmentWithout``` powinno być tak ```isNoEquipContained```. W ogóle, wydaje mi się że taka nazwa to bardziej pasuje do wartości ```boolean```, a sama lambda powinna wyglądać trochę inaczej.